slapo/sharding/shard_ops.py

```diff
@@ -548,7 +548,7 @@ class ShardConv1D(ShardMethod):
     and returns the output type (partial or partition) after sharding.
     """
 
-    class Conv1DWithSyncFunc(Conv1D):
+    class Conv1DWithSyncFunc(Conv1D if Conv1D is not None else nn.Module):
```

Comment:
If transformers is not available we should not run this op anyways, so I would suggest raising an exception in its constructor.

Reply:
Fixed